lib/code_fund/stats/user_impressions.ex

```diff
@@ -0,0 +1,103 @@
+defmodule CodeFund.Stats.UserImpressions do
+  use GenServer
+  use CodeFundWeb, :query
+  alias CodeFund.Schema.UserImpression
+
+  @refresh_interval :timer.minutes(60)
+
+  defmodule State do
+    defstruct impression_count: 0, click_count: 0, click_rate: 0.0, distribution_amount: 0.0
+  end
+
+  # Client API
+
+  def last_thirty_days() do
+    GenServer.call(__MODULE__, :stats)
+  end
+
+  # Invoked by the Supervisor process
+  def start_link do
+    GenServer.start_link(__MODULE__, %State{}, name: __MODULE__)
+  end
+
+  # Inside the spawned GenServer process
+  def init(state) do
+    schedule_refresh(:timer.seconds(5))
+    {:ok, state}
+  end
+
+  def handle_call(:stats, _from, state) do
+    {:reply, state, state}
+  end
+
+  def handle_info(:refresh, _state) do
```

Comment:
I think testing these should be pretty straightforward since you should be able to ask the system where this process is in a test and send it messages. i.e. - send a call, assert it replies, stub a `user_impression`, send it a refresh message, send a message to `stats` call and assert the numbers have changed in the pid state accordingly.

Reply:
I setup a very basic test but have had issues with inserting records into `user_impressions` as its a materialized view which doesn't allow direct updates. I pulled those efforts out. Could use some help on the other types of tests you'd like to see on this.

Reply:
Tried running the SQL query to refresh the view within the test; however, tests run in their own transaction so the data never commits and the view never refreshes since no data is committed.

I changed the strategy to simply verify that sending a signal to the gen server triggers the refresh.